### PR TITLE
Port to dune

### LINKED ISSUE #30: jbuilder/dune support?
Can I build programs using ppx_deriving_cmdliner with jbuilder? Adding   

(preprocess (pps (ppx_deriving_cmdliner)))

does not seem to work.

### PR DESCRIPTION
**Pull request by constfun**
I mostly followed the patterns established in https://github.com/ocaml-ppx/ppx_deriving_yojson

Also, made some changes to test and travis setup to ensure that both still pass.

And, added ocaml-version > 4.03 constraint to opam file since project failed to build under 4.03.

`make all-supported-ocaml-versions` is in the Makefile now and works locally assuming you have the local switches setup and dependencies installed.

Fixes #30

### CODE REVIEW

File: pkg/pkg.ml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#use "topfind"
+#require "topkg-jbuilder.auto"
```
Comment (smondet) [inline pkg/pkg.ml]: Why is this file still there?

**Comment by ihodes** [comment]
This is great, thank you for the PR!

**Comment by smondet** [comment]
@constfun ping?

File: pkg/pkg.ml (hunk 1 above)
Comment (constfun) [reply pkg/pkg.ml]: I'm not familiar with `topkg` and to err on side of caution chose to use the analogue from https://github.com/ocaml-ppx/ppx_deriving_yojson/blob/master/pkg/pkg.ml which looked like the jbuilder version of it.

**Comment by constfun** [comment]
It appears my GH notifications were borked, sorry for the slow reply here.

File: pkg/pkg.ml (hunk 1 above)
Comment (rgrinberg) [reply pkg/pkg.ml]: This is only for packages which use topkg-jbuilder. topkg-jbuilder has been superseded by dune-release which requires no boilerplate files.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,6 @@
 _build
 *.install
 pkg/META
+.merlin
+dune-project
```
[the remark this answers is not in the document]
Comment (smondet) [reply .gitignore]: Indeed , thanks → https://github.com/hammerlab/ppx_deriving_cmdliner/pull/33

**Pull request merged**